core/src/main/java/io/github/fourlastor/game/coordinates/Hex.java

```diff
@@ -65,4 +64,67 @@ public void set(Hex other) {
     public boolean sameAxisAs(Hex hex) {
         return hex.cube.x == cube.x || hex.cube.y == cube.y || hex.cube.z == cube.z;
     }
+
+    public Hex offset(Direction direction, int amount) {
+        return new Hex(cube.cpy().add(direction.cube.cpy().scl(amount)));
+    }
+
+    public enum Direction {
+        S(0, -1, 1),
+        SE(1, -1, 0),
+        NE(1, 0, -1),
+        N(0, +1, -1),
+        NW(-1, 1, 0),
+        SW(-1, 0, 1),
+        NONE(0, 0, 0),
+        ;
+
+        private final GridPoint3 cube;
+
+        Direction(int x, int y, int z) {
+            cube = new GridPoint3(x, y, z);
+        }
+
+        public static Direction byValue(GridPoint3 cube) {
+            return Arrays.stream(values())
+                    .filter(dir -> dir.cube.equals(cube))
+                    .findFirst()
+                    .get();
+        }
+
+        public static Direction byName(String value) {
+
+            Optional<Direction> test = Arrays.stream(values())
+                    .filter(dir -> dir.name().equalsIgnoreCase(value))
+                    .findFirst();
+            return test.orElse(null);
+        }
+
+        public static Direction byValue(int x, int y, int z) {
+            return byValue(new GridPoint3(x, y, z));
+        }
+
+        public boolean equals(Direction direction) {
+            return cube.equals(direction.cube);
+        }
```

Comment:
this is not the correct signature for equals:
```java
    @Override
    public boolean equals(Object o) {
        if (this == o) return true;
        if (o == null || getClass() != o.getClass()) return false;

        Hex hex = (Hex) o;

        if (!offset.equals(hex.offset)) return false;
        return cube.equals(hex.cube);
    }
```

I suggest letting the editor generate it for you as they're trickier than they look (we'll get rid of these with kotlin since it does it for you :grimacing: )

Reply:
actually, I think you don't need an equals at all, there are only N instances so the equality by instance comparison (default) works